src/preact/base-element.js

```diff
@@ -321,12 +339,23 @@ function collectProps(Ctor, element, defaultProps) {
   // as separate properties. Thus in a carousel the plain "children" are
   // slides, and the "arrowNext" children are passed via a "arrowNext"
   // property.
+  const errorMessage =
+    'only one of "passthrough", "passthroughNonEmpty"' +
+    ' or "children" may be given';
   if (passthrough) {
-    devAssert(
-      !childrenDefs,
-      'only one of "passthrough" or "children" may be given'
-    );
+    devAssert(!childrenDefs && !passthroughNonEmpty, errorMessage);
     props['children'] = [<Slot />];
+  } else if (passthroughNonEmpty) {
+    devAssert(!childrenDefs, errorMessage);
+    // If all children are whitespace text nodes, consider the element as
+    // having no children
+    props['children'] = element
+      .getRealChildNodes()
+      .every(
```

Comment:
`every` doesn't work for a case of empty array, does it?

Reply:
It seems to be working.  If it's an empty array, the `every` will return `true`.

Reply:
Cool.